### PR TITLE
Catch exceptions correctly in server.cpp

### PR DESCRIPTION
**Pull request by 0wwafa**
Catched exceptions in server load.

### CODE REVIEW

**Review by mofosyne — APPROVED** [review]
I like that you removed the redundant else statement good catch.

Anyhow looks good to me. Let's wait for the check to pass and then merge.

**Comment by 0wwafa** [comment]
Why sis one check fail?

**Comment by 0wwafa** [comment]
LOL
Run editorconfig-checker
examples/server/server.cpp:
	3038: Trailing whitespace

File: examples/server/server.cpp [hunk 1]
```diff
@@ -3028,13 +3028,25 @@ int main(int argc, char ** argv) {
...
         state.store(SERVER_STATE_READY);
+
+    } catch (const std::exception &e) {
+        // Catch known standard exceptions
+        LOG_ERROR("unable to load model", {{"error", e.what()}});
+        exit(1);
+    } catch (...) {
+        // Catch all other exceptions
+        LOG_ERROR("unable to load model", {{"error", "Unknown Exception"}});
+        exit(1);
```
Comment (ngxson) [inline examples/server/server.cpp:3049]: IMO it's quite strange to me to call `exit()` here, knowing that earlier we use `return 1;`

Comment (0wwafa) [reply examples/server/server.cpp:3049]: do you want me to change that?

**Comment by 0wwafa** [comment]
why is this still not accepted?

**Comment by ngxson** [comment]
After a second look on the PR, I don't think it's necessary to add this try..catch block. If there is a problem loading the model, `ctx_server.load_model(params)` already returns false. Instead, your try..catch mostly to wrap `ctx_server.init()`

In addition, it would be nice to have a concrete example of a model failed to load (i.e. example of model / command). It makes no sense to fix a problem that never occurs.

File: examples/server/server.cpp [hunk 2]
```diff
@@ -3028,13 +3028,25 @@ int main(int argc, char ** argv) {
...
+            return 1;
+        }
+        // Initialize the server context if the model is loaded successfully
         ctx_server.init();
         state.store(SERVER_STATE_READY);
+
+    } catch (const std::exception &e) {
+        // Catch known standard exceptions
+        LOG_ERROR("unable to load model", {{"error", e.what()}});
+        exit(1);
```
Comment (ngxson) [inline examples/server/server.cpp:3045]: You forgot to change this.

File: examples/server/server.cpp [hunk 3]
```diff
@@ -3028,13 +3028,25 @@ int main(int argc, char ** argv) {
...
 
-    // load the model
-    if (!ctx_server.load_model(params)) {
-        state.store(SERVER_STATE_ERROR);
-        return 1;
-    } else {
+    try {
+        // Attempt to load the model
+        if (!ctx_server.load_model(params)) {
+            LOG_ERROR("unable to load model", {{"error", "Load model failed"}});
```
Comment (ngxson) [inline examples/server/server.cpp:3034]: Here is why it makes no sense to me: we already catch the `unable to load model`. Why do we need another try..catch to do that?

**Review by ngxson — CHANGES_REQUESTED** [review]

File: examples/server/server.cpp (hunk 2 above)
Comment (0wwafa) [reply examples/server/server.cpp:3056]: ok.. I give up... 3 days and still nothing for a simple modification..
also: on windows, the program crashes for every stupid error like putting a non existant model filename.
that just plain wrong.
all exceptions should be caught. mine was only an example.

Comment (0wwafa) [reply examples/server/server.cpp:3056]: image

**Comment by mofosyne** [comment]
Everyone is a volunteer here. We do what we can. My understanding is that you thought `merging soon` means it will definitely be merged in. On second assessment I can see that the label really should be labeled as `merge ready` and is there to give other people in the team time to assess and provide feedback.

Anyway we appreciate the fact that you made an effort and will note down your recommendation separately as you have deleted this patch branch.

**Pull request closed without merging**